disnake/onboarding.py

```diff
@@ -0,0 +1,194 @@
+    prompts: List[:class:`OnboardingPrompt`]
+        The prompts shown during onboarding and in community customization.
+    enabled: :class:`bool`
+        Whether onboarding is enabled.
+    default_channel_ids: FrozenSet[:class:`int`]
+        The IDs of the channels that will automatically be shown to new members.
+    """
+
+    __slots__ = (
+        "guild",
+        "prompts",
+        "enabled",
+        "default_channel_ids",
+    )
+
+    def __init__(self, *, guild: Guild, data: OnboardingPayload):
+        self.guild = guild
+        self._from_data(data)
+
+    def _from_data(self, data: OnboardingPayload):
+        self.prompts = [
+            OnboardingPrompt(data=prompt, guild=self.guild) for prompt in data["prompts"]
+        ]
+        self.enabled = data["enabled"]
+        self.default_channel_ids = (
+            frozenset(map(int, exempt_channels))
+            if (exempt_channels := data["default_channel_ids"])
+            else frozenset()
+        )
+
+    def __repr__(self) -> str:
+        return (
+            f"<Onboarding guild={self.guild!r} prompts={self.prompts!r} enabled={self.enabled!r}>"
+        )
+
+    @property
+    def default_channels(self) -> List[GuildChannel]:
+        """List[:class:`abc.GuildChannel`]: The list of channels that will be shown to new members by default."""
+        return list(filter(None, map(self.guild.get_channel, self.default_channel_ids)))
+
+
+class OnboardingPrompt(Hashable):
+    """Represents an onboarding prompt.
+
+    .. versionadded:: 2.9
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The onboarding prompt's ID.
+    type: :class:`OnboardingPromptType`
+        The onboarding prompt's type.
+    options: List[:class:`OnboardingPromptOption`]
+        The onboarding prompt's options.
+    title: :class:`str`
+        The onboarding prompt's title.
+    single_select: :class:`bool`
+        Whether users are limited to selecting one option for the prompt.
+    required: :class:`bool`
+        Whether the prompt is required before a user completes the onboarding flow.
+    in_onboarding: :class:`bool`
+        Whether the prompt is present in the onboarding flow.
+        If ``False``, the prompt will only appear in community customization.
+    """
+
+    __slots__ = (
+        "guild",
+        "id",
+        "title",
+        "options",
+        "single_select",
+        "required",
+        "in_onboarding",
+        "type",
+    )
+
+    def __init__(self, *, guild: Guild, data: OnboardingPromptPayload):
+        self.guild = guild
+
+        self.id = int(data["id"])
+        self.title = data["title"]
+        self.options = [
+            OnboardingPromptOption(data=option, guild=guild) for option in data["options"]
+        ]
+        self.single_select = data["single_select"]
+        self.required = data["required"]
+        self.in_onboarding = data["in_onboarding"]
+        self.type = try_enum(OnboardingPromptType, data["type"])
```

Comment:
could you add type annotations to these, as well as the ones in the other two classes?

Reply:
Alright!

Reply:
d9d0c00772c29d379e72a40b9711581a36588b8b